lib/matplotlib/markers.py

```diff
@@ -142,6 +143,233 @@
+    CARETRIGHTBASE: BoxSides(_caret_side, _caret_side, _caret_bottom,
+                             _caret_tip),
+    CARETUPBASE: BoxSides(_caret_tip, _caret_bottom, _caret_side, _caret_side),
+    CARETDOWNBASE: BoxSides(_caret_bottom, _caret_tip, _caret_side,
+                            _caret_side),
+    '': BoxSides(None, None, None, None),
+    ' ': BoxSides(None, None, None, None),
+    'None': BoxSides(None, None, None, None),
+    None: BoxSides(None, None, None, None),
+}
+
+
+def _get_padding_due_to_angle(width, path_end_angle, joinstyle='miter',
+                              capstyle='butt'):
+    """Computes how much a stroke of *width* overflows the naive bbox at a
+    corner described by *path_end_angle*.
+
+    Parameters
+    ----------
+    width : float
+        `markeredgewidth` used to draw the stroke that we're computing the
+        overflow of
+    path_end_angle : PathEndAngle
+        precomputed property of a corner that allows us to compute the overflow
+
+    Returns
+    -------
+    pad : float
+        amount of overflow
+
+    """
+    if path_end_angle is None or width < 0:
+        return np.nan
+    phi, theta = path_end_angle.incidence_angle, path_end_angle.corner_angle
+    if theta is not None and (theta < 0 or theta > np.pi) \
+            or phi < 0 or phi > np.pi:
+        raise ValueError("Corner angles should be in [0, pi].")
+    if phi > np.pi/2:
+        # equivalent by symmetry, but keeps math simpler
+        phi = np.pi - phi
+    # if there's no corner (i.e. the path just ends, as in the "sides" of the
+    # carets or in the non-fillable markers, we can compute how far the outside
+    # edge of the markeredge stroke extends outside of the bounding box of its
+    # path using the law of sines: $\sin(\phi)/(w/2) = \sin(\pi/2 - \phi)/l$
+    # for $w$ the `markeredgewidth`, $\phi$ the incidence angle of the line,
+    # then $l$ is the length along the outer edge of the stroke that extends
+    # beyond the bouding box. We can translate this to a distance perpendicular
+    # to the bounding box E(w, \phi) = l \sin(\phi)$, for $l$ as above.
+    if theta is None:
+        # also note that in this case, we shouldn't check _joinstyle because
+        # it's going to be "round" by default but not actually be in use. what
+        # we care about is _capstyle, which is currently always its default
+        # "butt" for all markers. if we find otherwise, we should change this
+        # code to check for the "projecting" and "round" cases
+        if capstyle != 'butt':
+            raise NotImplementedError("Only capstyle='butt' currently needed")
+        pad = (width/2) * np.cos(phi)
+    # the two "same as straight line" cases are NaN limits in the miter formula
+    elif np.isclose(theta, 0) and np.isclose(phi, 0) \
+    or np.isclose(theta, np.pi) and np.isclose(phi, np.pi/2):
+        pad = width/2
+    # to calculate the offset for _joinstyle == 'miter', imagine aligning the
```

Comment:
If you're really going to go through all this (I agree with the other maintainers that this looks slightly overkill...), you may also want to take miter_limits into account -- see https://github.com/matplotlib/matplotlib/issues/9830.